iso15118/secc/controller/ev_data.py

```diff
@@ -0,0 +1,152 @@
+    PVEVMaxPowerLimit,
+    PVEVMaxVoltage,
+    PVEVMaxVoltageLimit,
+)
+from iso15118.shared.messages.din_spec.datatypes import ChargeService
+from iso15118.shared.messages.enums import AuthEnum
+from iso15118.shared.messages.iso15118_20.ac import (
+    ACChargeParameterDiscoveryReqParams,
+    BPTACChargeParameterDiscoveryReqParams,
+    BPTDynamicACChargeLoopReqParams,
+    BPTScheduledACChargeLoopReqParams,
+    DynamicACChargeLoopReqParams,
+    ScheduledACChargeLoopReqParams,
+)
+from iso15118.shared.messages.iso15118_20.dc import (
+    BPTDCChargeParameterDiscoveryReqParams,
+    BPTDynamicDCChargeLoopReqParams,
+    BPTScheduledDCChargeLoopReqParams,
+    DCChargeParameterDiscoveryReqParams,
+    DynamicDCChargeLoopReqParams,
+    ScheduledDCChargeLoopReqParams,
+)
+
+
+@dataclass
+class EVDataContext:
+    dc_current_request: Optional[int] = None
+    dc_voltage_request: Optional[int] = None
+    ac_current: Optional[dict] = None  # {"l1": 10, "l2": 10, "l3": 10}
+    ac_voltage: Optional[dict] = None  # {"l1": 230, "l2": 230, "l3": 230}
+    soc: Optional[int] = None  # 0-100
+    departure_time: Optional[int] = None
+
+    remaining_time_to_full_soc_s: Optional[int] = None
+    remaining_time_to_bulk_soc_s: Optional[int] = None
+    evcc_id: Optional[str] = None
+
+    # Common to both ISO15118-20 AC and DC
+    ev_max_charge_power: float = 0.0
+    ev_min_charge_power: float = 0.0
+
+    # Common to both ISO15118-20 AC-BPT and DC-BPT
+    ev_max_discharge_power: Optional[float] = None
+    ev_min_discharge_power: Optional[float] = None
+
+    # Specific to ISO 15118-20 AC
+    ev_max_charge_power_l2: Optional[float] = None
+    ev_max_charge_power_l3: Optional[float] = None
+    ev_min_charge_power_l2: Optional[float] = None
+    ev_min_charge_power_l3: Optional[float] = None
+
+    # Specific to ISO 15118-20 AC BPT
+    ev_max_discharge_power_l2: Optional[float] = None
+    ev_max_discharge_power_l3: Optional[float] = None
+    ev_min_discharge_power_l2: Optional[float] = None
+    ev_min_discharge_power_l3: Optional[float] = None
+
+    # Specific to ISO 15118-20 DC
+    ev_max_charge_current: Optional[float] = None
+    ev_min_charge_current: Optional[float] = None
+    ev_max_voltage: Optional[float] = None
+    ev_min_voltage: Optional[float] = None
+    target_soc: Optional[int] = None
+
+    # Specific to ISO 15118-20 DC BPT
+    ev_max_discharge_current: Optional[float] = None
+    ev_min_discharge_current: Optional[float] = None
+
+    ev_target_energy_request: Optional[float] = None
+    ev_max_energy_request: Optional[float] = None
+    ev_min_energy_request: Optional[float] = None
+    ev_present_active_power: Optional[float] = None
+    ev_present_active_power_l2: Optional[float] = None
+    ev_present_active_power_l3: Optional[float] = None
+    ev_present_reactive_power: Optional[float] = None
+    ev_present_reactive_power_l2: Optional[float] = None
+    ev_present_reactive_power_l3: Optional[float] = None
```

Comment:
this exists but only in the context dynamic AC charging loop

Reply:
Yes, have now added a comment to clarify this.